core/src/Kokkos_View.hpp

```diff
@@ -450,6 +450,64 @@ struct ViewTraits {
 template <class DataType, class... Properties>
 class View;
 
+template <class T1, class T2>
+struct is_always_assignable : public std::false_type {};
```

Comment:
Minor: I would not make all of the non-`View` partial specializations `false`; it just makes it too easy for someone to get a false negative for something like `is_always_assignable_v<View<double*>, View<double*> const&>` (which is really easy to accidentally do in generic contexts with forwarding references). Instead, just leave it undefined so that the user gets a compilation error.

```suggestion
template <class T1, class T2>
struct is_always_assignable;
```

Reply:
I think you are pointing out a defect.  I expect `is_always_assignable_v<View<double*>, View<double*> const&>` should be `true`.  We should remove the cvref qualifier on the rhs.